backend/jobsearch/src/main/kotlin/com/jobsearch/controller/VacancyController.kt

```diff
@@ -13,41 +16,122 @@ import org.springframework.web.bind.annotation.*
+        val dataBody = vacancyService.updateVacancy(vacancyId, vacancyDto)
+        return mapResponseEntity(dataBody)
     }
 
     @DeleteMapping("{id}")
-    @ResponseStatus(HttpStatus.NO_CONTENT)
-    fun deleteVacancy(@PathVariable("id") vacancyId: Int): String {
-        return vacancyService.deleteVacancy(vacancyId)
+    fun deleteVacancy(@PathVariable("id") vacancyId: Int): ResponseEntity<StandardResponse<Unit>> {
+        val dataBody = vacancyService.deleteVacancy(vacancyId)
+        return mapResponseEntity(dataBody)
     }
 
-    @GetMapping("/search")
-    fun searchVacancies(
-        @RequestParam(required = false) salary: Int?,
-        @RequestParam(required = false) jobFamilyId: Int?,
-        @RequestParam(required = false) yearsOfExperience: Int?
-    ): List<VacancyDto> {
+    /**
+     * Maps a response entity with the provided data body and status.
+     * Returns a ResponseEntity of StandardResponse of VacancyResponseDTO.
+     *
+     * @param dataBody The data body to be included in the response.
+     * @param status The HTTP status code (default is HttpStatus.OK).
+     * @return ResponseEntity<StandardResponse<VacancyResponseDTO>> The mapped response entity.
+     */
+    fun mapResponseEntity(
+        dataBody: VacancyResponseDTO,
+        status: HttpStatus = HttpStatus.OK): ResponseEntity<StandardResponse<VacancyResponseDTO>>{
+
+        val bodyContent = StandardResponse(
+            status = status.value(),
+            data = dataBody,
+            message = status.reasonPhrase
+        )
+        return ResponseEntity
+            .status(status)
+            .body(bodyContent)
+    }
+
+
+    /**
+     * Maps the response entity for a successful deletion.
+     *
+     * @param deletedBody The body of the response.
+     * @param status The HTTP status of the response. Default is HttpStatus.NO_CONTENT.
+     * @return The ResponseEntity containing the mapped response entity.
+     */
+    fun mapResponseEntity(
+        deletedBody: Unit,
+        status: HttpStatus = HttpStatus.NO_CONTENT): ResponseEntity<StandardResponse<Unit>>{
+
+        val bodyContent = StandardResponse(
+            status = status.value(),
+            data = deletedBody,
+            message = status.reasonPhrase
+        )
+        return ResponseEntity
+            .status(status)
+            .body(bodyContent)
+    }
 
-        return vacancyService.findVacanciesByFilter(salary, jobFamilyId, yearsOfExperience)
+    /**
+     * Maps a response entity with the provided data body list and status.
+     * Returns a ResponseEntity of StandardResponse of VacancyResponseDTO.
+     *
+     * @param dataBodyList The data body list to be included in the response.
+     * @param status The HTTP status code (default is HttpStatus.OK).
+     * @return ResponseEntity<StandardResponse<List<VacancyResponseDTO>> The mapped response entity.
+     */
+    fun mapResponseEntity(
```

Comment:
all `mapResponseEntity` methods have the same logic
Please create one generic method and use it for all situations
And please make it `private`